aten/src/ATen/native/cuda/Dropout.cu

```diff
@@ -0,0 +1,159 @@
+#include "ATen/ATen.h"
+#include "ATen/AccumulateType.h"
+#include "ATen/cuda/CUDAApplyUtils.cuh"
+#include "detail/IndexUtils.cuh"
+#include "detail/TensorInfo.cuh"
+#include "curand_kernel.h"
+
+#include <THC/THCGeneral.h>
+#include <THC/THCTensorRandom.h>
+#include <THC/THCGenerator.hpp>
+
+
+THCGenerator* THCRandom_getGenerator(THCState* state);
+
+namespace at{
+namespace native{
+
+namespace {
+
+//due to limitations of philox generator UNROLL has to be 4
```

Comment:
Can you please comment what particular limitations do you have in mind in here?

Reply:
Philox generated 128 random bits at a time, so to use them all, the unroll has to be 4. I'll put it in a comment.